### ISSUE TITLE
cant get it to go beyond converting nessus file to csv

### ISSUE THREAD

**Issue by rshrestha6** [HASecuritySolutions/VulnWhisperer#41]
Hi,
Still cant get it to do anything beyond converting the nessus file to csv,  both logstash and kibana service status shows running ok,  not sure what am i doing wrong but it doesn't generate report and/or output the visualization to kibana, I have followed the video sample to follow the exact steps but I am not getting me anywhere. 

Please advise

below is the output and attached are the logstash yml, logstash conf and .ini file
[logstash_conf.txt](https://github.com/austin-taylor/VulnWhisperer/files/1823474/logstash_conf.txt)
[logstash_yml.txt](https://github.com/austin-taylor/VulnWhisperer/files/1823476/logstash_yml.txt)
[ini file.txt](https://github.com/austin-taylor/VulnWhisperer/files/1823477/ini.file.txt)

OUTPUT
**test@test-virtual-machine:~/Desktop/VulnWhisperer-master$ sudo vuln_whisperer -c configs/frameworks_example.ini -s nessus
[INFO] Connected to database at /opt/vulnwhisp/database/report_tracker.db
[INFO] Attempting to connect to nessus...
[SUCCESS] Connected to nessus on localhost:8834
[INFO] Gathering all scan data... this may take a while...
[INFO] Identified 9 scans to be processed
[INFO] Directory already exist for /opt/vulnwhisp/nessus/Trash - Skipping creation
[INFO] Directory already exist for /opt/vulnwhisp/nessus/My Scans - Skipping creation
Download for file id 1168345030.
.
Processing 1/9 for scan: test
[INFO] 78 records written to test_25_26_1521411373.csv 
Download for file id 819727134.
.
Processing 2/9 for scan: scan
[INFO] 86 records written to scan_22_23_1520385708.csv 
Download for file id 1969498333.
.
Processing 3/9 for scan: attemptthird
[INFO] 79 records written to attemptthird_19_20_1519954223.csv 
Download for file id 79218840.
.
Processing 5/9 for scan: attempt 1
[INFO] 79 records written to attempt_1_14_15_1519952202.csv 
Download for file id 950718717.
.
Processing 8/9 for scan: test
[INFO] 79 records written to test_7_8_1519946993.csv**

**Comment by austin-taylor** [comment]
Looks like the csv's are being written to disk. Are you trying to setup ELK now?

**Comment by splendid12** [comment]
I already have the ELK stack setup, have logstash, kibana and elasticsearch all running, not sure why it it only writing the csv file into the loacal disk (/opt/vulnwhisp/nessus/My Scans folder) and sending it to elasticsearch.

**Comment by austin-taylor** [comment]
Can you post your logstash.conf file?

**Comment by austin-taylor** [comment]
also could you please post the permissions you have applied to /opt/vulnwhisp/nessus

**Comment by splendid12** [comment]
permissions

test@test-virtual-machine:~$ ls -l  /opt/vulnwhisp/nessus
total 8
drwxr-xr-x 2 root root 4096 Mar 18 18:33 My Scans
drwxr-xr-x 2 root root 4096 Mar 18 18:33 Trash

conf attached
[logstash_conf.txt](https://github.com/austin-taylor/VulnWhisperer/files/1845932/logstash_conf.txt)
 

thanks

**Comment by DavidJWilliamson** [comment]
I am having the same exact issue. I cannot get logstash to go into elastic.

**Comment by austin-taylor** [comment]
https://github.com/austin-taylor/VulnWhisperer/blob/master/logstash/1000_nessus_process_file.conf#L10

You need to either update that line to process /opt/vulnwhisp/nessus or update your .ini file to process that directory

**Comment by austin-taylor** [comment]
I've updated the basic config file to align with the logstash config files. Please let me know if that resolve it.

**Comment by splendid12** [comment]
Thanks.

I'm getting the following error now.

python vulnwhisp.py -c /home/test/Desktop/VulnWhisperer-master/configs/example.ini -s nessus
Traceback (most recent call last):
  File "vulnwhisp.py", line 8, in <module>
    from frameworks.openvas import OpenVAS_API
  File "/home/test/Desktop/VulnWhisperer-master/vulnwhisp/frameworks/openvas.py", line 12, in <module>
    from ..utils.cli import bcolors
ValueError: Attempted relative import beyond toplevel package

**Comment by splendid12** [comment]
Thanks but it is still the same, not sure what I am doing wrong..    :(

**Comment by DavidJWilliamson** [comment]
I also found that path difference earlier today before commenting, but it did not help.

**Comment by austin-taylor** [comment]
Did you restart logstash after updating the config? service logstash restart

**Comment by DavidJWilliamson** [comment]
Yes I did and it still was not working. I was able to finally get it working once I removed all other logstash configs except for the nessus one. Maybe it was just my error, but I do not remember seeing anything to say only have the one in the conf directory.

**Comment by splendid12** [comment]
Hi Dwill,

which config file did u edited,  was it the .ini file? also were you able to get the data on to elasticsearch?

**Comment by splendid12** [comment]
Hi austin,

I was finally able to get it to working too, think logstash was pointing to a different config file.
Just couple of things, I noticed when I process new scan files from nessus it also looks for historical data in the TRASH folder,

_Processing 1/5 for scan: test
[FAIL] ERROR: [Errno 2] No such file or directory: u'/opt/vulnwhisperer/nessus/Trash/test_25_26_1521411373.csv'_

it appears to creates cache of all the historical scans into the TRASH folder, I think I accidentally deleted the TRASH folder and it wasn't working until  I manually created the TRASH folder again and the error went away, not sure what its doing but is it always indexing the historical nessus scans (or .csv file) from the TRASH folder too, if yes then how do I get it to only reference the latest nessus data.

Also it does not work with imported scan, I remember you mentioned that it looks for some metadata that the python is looking for to pull the scans in. is there a fix for this? would be absolutely great if we could process imported scans. 


Thanks for all your help..
